hack/e2e-setup-gpu-cluster.sh

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This shell is used to setup Kind cluster for Kubeflow Trainer e2e tests.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -x
+
+# Configure variables.
+KIND=${KIND:-./bin/kind}
+K8S_VERSION=${K8S_VERSION:-1.32.0}
+GPU_OPERATOR_VERSION="v25.3.2"
+KIND_NODE_VERSION=kindest/node:v${K8S_VERSION}
+NAMESPACE="kubeflow-system"
+TIMEOUT="5m"
+
+# Kubeflow Trainer images.
+# TODO (andreyvelich): Support initializers images.
+CONTROLLER_MANAGER_CI_IMAGE_NAME="ghcr.io/kubeflow/trainer/trainer-controller-manager"
+CONTROLLER_MANAGER_CI_IMAGE_TAG="test"
+CONTROLLER_MANAGER_CI_IMAGE="${CONTROLLER_MANAGER_CI_IMAGE_NAME}:${CONTROLLER_MANAGER_CI_IMAGE_TAG}"
+echo "Build Kubeflow Trainer images"
+sudo docker build . -f cmd/trainer-controller-manager/Dockerfile -t ${CONTROLLER_MANAGER_CI_IMAGE}
+
+# Set up Docker to use NVIDIA runtime.
+sudo nvidia-ctk runtime configure --runtime=docker --set-as-default --cdi.enabled
+sudo nvidia-ctk config --set accept-nvidia-visible-devices-as-volume-mounts=true --in-place
+sudo systemctl restart docker
+
+# Create a Kind cluster with GPU support.
+nvkind cluster create --image "${KIND_NODE_VERSION}"
+CLUSTER_NAME=$(kind get clusters | grep nvkind)
+nvkind cluster print-gpus
+
+# Install gpu-operator to make sure we can run GPU workloads.
+echo "Install NVIDIA GPU Operator"
+kubectl create ns gpu-operator
```

Comment:
Are you going to refactor this script in the followup PR ? E.g. we can just re-use this script, and create cluster with `nvkind` if GPU is used: https://github.com/kubeflow/trainer/blob/05cf8cf97e2a5318d960f2c510ff4785e227b402/hack/e2e-setup-cluster.sh#L33-L37

Reply:
Yes @andreyvelich In future, as discussed, I will merge this into single [e2e-setup-cluster.sh](https://github.com/kubeflow/trainer/blob/05cf8cf97e2a5318d960f2c510ff4785e227b402/hack/e2e-setup-cluster.sh#L33-L37) based on flag.

Reply:
@jaiakash Please create an issue and assign it to yourself to track it.

Reply:
Sure I will create a issue for this.

By the way, can you review issue: https://github.com/kubeflow/trainer/issues/2809
PR: https://github.com/kubeflow/trainer/pull/2810